extide/gradle/netbeans-gradle-tooling/src/main/java/org/netbeans/modules/gradle/tooling/NbProjectInfoBuilder.java

```diff
@@ -487,6 +501,15 @@ private void detectDependencies(NbProjectInfoModel model) {
 
             if (resolvable(it)) {
                 try {
+                    Set<ResolvedArtifact> arts = it.getResolvedConfiguration()
```

Comment:
The `collectArtifacts` [at line 516](https://github.com/apache/netbeans/pull/3942/files#diff-9fd2231c8ce99fda4d6d8295c69707bf67f91b6b492ee886a736f8a4a6d55a08R516) fetches modules recursively, but JARs are not. Therefore the `jvm_resolved_artifacts` map was incomplete: did not assign artifacts for all components turned into `ModuleDependency` for the `micronaut-core` project.
Adding this provides as much JAR locations as possible. Checked that does not need much additional time compared to original.